src/Compute/Compute/Manual/VirtualMachineScaleSetCreateOrUpdateMethod.cs

```diff
@@ -667,11 +672,20 @@ async Task SimpleParameterSetExecuteCmdlet(IAsyncCmdlet asyncCmdlet)
                     this.EnableVtpm = true;
                 }
             }
-            
+            // default Win2022AzureEdition img for explicitly set Standard.
+            // handles when default img was set for ImageName parameter.
+            if (this.IsParameterBound(c => c.SecurityType)
+                && this.SecurityType?.ToLower() == ConstantValues.StandardSecurityType
+                && !this.IsParameterBound(c => c.ImageName)
+                && !this.IsParameterBound(c => c.ImageReferenceId)
+                && !this.IsParameterBound(c => c.SharedGalleryImageId))
+            {
+                this.ImageName = ConstantValues.TrustedLaunchDefaultImageAlias;
+            }
+
             // API does not currently support Standard securityType value, so need to null it out here. 
             if (this.IsParameterBound(c => c.SecurityType)
-                && this.SecurityType != null
-                && this.SecurityType.ToString().ToLower() == ConstantValues.StandardSecurityType)
+                && this.SecurityType?.ToLower() == ConstantValues.StandardSecurityType)
             {
                 this.SecurityType = null;
```

Comment:
this seems unintuitive. does the API still not support Standard security type? why is that? do they plan to update this?

Reply:
The API does not support Standard. IT is a dummy value for the client to use to opt out of the defaulting TL logic.